presto-main/src/main/java/com/facebook/presto/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -237,10 +250,12 @@ public void enqueue(List<SerializedPage> pages)
     }
 
     @Override
-    public void enqueue(int partition, List<SerializedPage> pages)
+    public void enqueue(Lifespan lifespan, int partition, List<SerializedPage> pages)
     {
+        requireNonNull(lifespan, "lifespan is null");
         checkState(partition == 0, "Expected partition number to be zero");
-        enqueue(pages);
+        checkState(lifespanCompletionCallback != null, "lifespanCompletionCallback has to be set before starting enqueue data");
```

Comment:
Maybe `lifespanCompletionCallback has to be set before starting to enqueue data`? (same for other places where this message is used)